shared/middleware/helpers/firebase_helper.rb

```diff
@@ -26,19 +26,48 @@ def table_as_csv(table_name)
     table_to_csv(records, column_order: ['id'])
   end
 
+  def number?(num)
+    !Float(num).nil? rescue false
+  end
+
+  def csv_as_table(csv_data)
+    records = {}
+    id = 1
+    table = CSV.parse(csv_data, headers: true)
+    table.each do |row|
+      record = OpenStruct.new
+      record.id = id
+      table.headers.each do |col|
+        value = (number? row[col]) ? row[col].to_f : row[col]
+        record[col] = value
+      end
+      records[id] = record.to_h.to_json
+      id += 1
+    end
+    # add id as first column
+    columns = table.headers.unshift('id')
```

Comment:
can we do this before we iterate over the table on line 37? or is there a reason this needs to come after?

Reply:
We have to set the `id` of each record based on our `id` counter (line 39), while every other column's value is set in the loop (line 42), so I think it makes sense to iterate through all the columns first, then add id to the list of columns at the end, but lmk if there's a better approach!